proto/spire/plugin/server/keymanager/v1/keymanager.proto

```diff
@@ -1,101 +1,130 @@
 syntax = "proto3";
-package spire.server.keymanager;
-option go_package = "github.com/spiffe/spire/proto/spire/server/keymanager";
+package spire.plugin.server.keymanager.v1;
+option go_package = "github.com/spiffe/spire-plugin-sdk/proto/spire/plugin/server/keymanager/v1;keymanagerv1";
 
-import "spire/common/plugin/plugin.proto";
+service KeyManager {
+    // Generates a new private key with the given ID. If a key already exists
+    // under that ID, it is overwritten. The fingerprint of the overwritten
+    // key is different than the original.
+    rpc GenerateKey(GenerateKeyRequest) returns (GenerateKeyResponse);
 
+    // Gets the public key information for the private key managed by the
+    // plugin with the given ID. If a key with the given ID does not exist,
+    // NOT_FOUND is returned.
+    rpc GetPublicKey(GetPublicKeyRequest) returns (GetPublicKeyResponse);
 
-enum KeyType {
-    UNSPECIFIED_KEY_TYPE = 0;
-    EC_P256 = 1;
-    EC_P384 = 2;
-    RSA_1024 = 3;
-    RSA_2048 = 4;
-    RSA_4096 = 5;
-}
+    // Gets all public key information for the private keys managed by the
+    // plugin.
+    rpc GetPublicKeys(GetPublicKeysRequest) returns (GetPublicKeysResponse);
 
-enum HashAlgorithm {
-    UNSPECIFIED_HASH_ALGORITHM = 0;
-    // These entries (and their values) line up with a subset of the go
-    // crypto.Hash constants
-    SHA224 = 4;
-    SHA256 = 5;
-    SHA384 = 6;
-    SHA512 = 7;
-    SHA3_224 = 10;
-    SHA3_256 = 11;
-    SHA3_384 = 12;
-    SHA3_512 = 13;
-    SHA512_224 = 14;
-    SHA512_256 = 15;
+    // Signs data with the private key with the given ID and fingerprint. If a
+    // key with the given ID does not exist or has a different fingerprint,
+    // NOT_FOUND is returned.
+    rpc SignData(SignDataRequest) returns (SignDataResponse);
 }
 
-
 message PublicKey {
+    // Required. The ID of the key, as provided when the key was created.
     string id = 1;
+
+    // Required. The type of the key.
     KeyType type = 2;
+
+    // Required. The public key data (PKIX encoded).
     bytes pkix_data = 3;
+
+    // Required. Fingerprint of the public key. The (id,fingerprint) tuple
+    // represents a unique identifier for the key. The fingerprint is used to
+    // ensure that a signing operation is taking place with the intended key
+    // (see SignData). Plugins are welcome to choose their fingerprinting
+    // algorithm. A naive implementation is a hash over the PKIX data.
```

Comment:
nit: a note here that this is not security critical, or a little bit more emphasis on it's "helper" status, would be good